port/unix/omrsysinfo.c

```diff
@@ -7560,10 +7564,21 @@ omrsysinfo_get_number_context_switches(struct OMRPortLibrary *portLibrary, uint6
 
 #if defined(LINUX)
 /*
- * A helper function to fully read a file.
+ * Reads the entire contents of a file descriptor into a dynamically growing buffer
+ *
+ * If the buffer is too small, it grows until all data is read
+ *
+ * @param[in] portLibrary Port library used for memory and file operations
+ * @param[in] file File descriptor to read from
+ * @param[in,out] data Pointer to buffer holding the file contents, reallocated if needed
+ * @param[in,out] size Pointer to buffer size in bytes, updated if buffer grows
+ *
+ * @return Total bytes read on success or a negative value on failure
+ *
+ * @note Caller must free the buffer using portLibrary->mem_free_memory()
```

Comment:
fixed grammar and punctuation
```suggestion
 * Reads the entire contents of a file descriptor into a dynamically growing buffer.
 *
 * If the buffer is too small, it grows until all data is read.
 *
 * @param[in] portLibrary Port library used for memory and file operations.
 * @param[in] file File descriptor to read from.
 * @param[in,out] data Pointer to the buffer holding the file contents; reallocated if needed.
 * @param[in,out] size Pointer to the buffer size in bytes; updated if the buffer grows.
 *
 * @return Total bytes read on success, or a negative value on failure.
 *
 * @note The caller must free the buffer using portLibrary->mem_free_memory().
```

Reply:
@babsingh , I have updated the change